lib/statsd.js

```diff
@@ -77,29 +80,15 @@ const Client = function (host, port, prefix, suffix, globalize, cacheDns, mock,
       this.dnsError = options.dnsError;
     }
     this.socket = options.socket;
-  } else if (options.cacheDns === true) {
-    dns.lookup(this.host, (err, address) => {
-      if (err === null) {
-        this.host = address;
-      } else {
-        this.dnsError = err;
-      }
-      this.socket = createTransport(this, {
-        host: this.host,
-        path: options.path,
-        port: this.port,
-        protocol: this.protocol,
-        stream: options.stream
-      });
-    });
-  } else {
-    if (options.useDefaultRoute) {
-      const defaultRoute = helpers.getDefaultRoute();
-      if (defaultRoute) {
-        console.log(`Got ${defaultRoute} for the system's default route`);
-        this.host = defaultRoute;
-      }
+  } else if (options.useDefaultRoute) {
+    const defaultRoute = helpers.getDefaultRoute();
+    if (defaultRoute) {
+      console.log(`Got ${defaultRoute} for the system's default route`);
+      this.host = defaultRoute;
     }
+  }
+
+  if (! this.socket) {
     this.socket = createTransport(this, {
```

Comment:
Go back to always creating the socket right away.  Putting it into a callback in the cacheDns case broke code